hub/integrations/cleanlab/cleanlab.py

```diff
@@ -0,0 +1,157 @@
+from typing import Any, Optional, Union
+import pandas as pd
+
+
+def clean_labels(
+    dataset: Any,
+    model: Any,
+    folds: int = 5,
+    pretrained: bool = False,
+    verbose: bool = True,
+    label_issues_kwargs: Optional[dict] = {},
+    label_quality_kwargs: Optional[dict] = {},
+):
+    """
+    Finds label errors in a dataset with cleanlab (github.com/cleanlab) open-source library.
+
+    Note:
+        The tensors used to obtain label_issues can be specified in `transofrm` or `tensors` when instantiating skorch module.
+        Setting `pretrained` to `True` is only recommended if there's no distribution shift in the dataset. in general, using cross-validation is strongly recommended.
+
+    Args:
+        dataset (class): Hub Dataset for training. The label issues will be computed for training set.
+        model (class): An instantiated skorch NeuralNet module.
+        folds (int): Sets the number of cross-validation folds used to compute out-of-sample probabilities for each example in the dataset. Only applicable if pretrained is `False`. The default is 5.
+        pretrained (bool): If using a pretrained model, set this to True. This will speed up the process by skipping cross-validation and obtaining predicted probabilities on a single `fit()`. Default is False.
+        verbose (bool): This parameter controls how much output is printed. Default is True.
+        label_issues_kwargs (dict, Optional): Keyword arguments to be passed to the `cleanlab.filter.find_label_issues` function. Options that may especially impact accuracy include: filter_by, frac_noise, min_examples_per_class. Default is `None`.
+        label_quality_kwargs (dict, Optional): Keyword arguments to be passed to the `cleanlab.rank.get_label_quality_scores` function. Options include: method, adjust_pred_probs. Default is `None`.
+
+    Returns:
+        pandas DataFrame of label issues for each example. Each row represents an example from the dataset and the DataFrame contains the following columns:
+            - label_issues: A boolean mask for the entire dataset where True represents a label issue and False represents an example that is confidently/accurately labeled.
+            - label_quality_scores: Label quality scores for each datapoint, where lower scores indicate labels less likely to be correct.
+            - predicted_labels: Class predicted by model trained on cleaned data for each example in the dataset.
+
+    Raises:
+        ...
+
+    """
+
+    from hub.integrations.cleanlab.label_issues import get_label_issues
+    from hub.integrations.common.utils import is_dataset
+
+    # Catch most common user errors early.
+    if not is_dataset(dataset):
```

Comment:
I think this can be a simple is instance check

Reply:
Initially, that's how I implemented it, but I'm getting `cannot import name 'Dataset' from partially initialized module 'hub.core.dataset'` when I try to `from hub.core.dataset import Dataset` in `skorch.py`.